From: Didier Raboud <odyx@debian.org>
Date: Sat, 1 Dec 2018 01:39:17 +0100
Subject: Mirror doc/developer/cleanwarnings.pl from upstream

Taken from upstream's master; it is needed to rebuild the doc PDFs from source
---
 doc/developer/cleanwarnings.pl | 41 +++++++++++++++++++++++++++++++++++++++++
 1 file changed, 41 insertions(+)
 create mode 100644 doc/developer/cleanwarnings.pl

diff --git a/doc/developer/cleanwarnings.pl b/doc/developer/cleanwarnings.pl
new file mode 100644
index 0000000..42df778
--- /dev/null
+++ b/doc/developer/cleanwarnings.pl
@@ -0,0 +1,41 @@
+#	-*- Mode: Perl -*-
+## Copyright (C) 2013 Robert Krawitz
+##
+## This program is free software; you can redistribute it and/or modify
+## it under the terms of the GNU General Public License as published by
+## the Free Software Foundation; either version 2, or (at your option)
+## any later version.
+##
+## This program is distributed in the hope that it will be useful,
+## but WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+## GNU General Public License for more details.
+##
+## You should have received a copy of the GNU General Public License
+## along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+# Remove annoying TeX warnings from db2pdf.
+
+$nc="";
+while (<>) {
+    next if /^$/;
+    chomp;
+  LINE:
+    if (/^Overfull \\hbox|^LaTeX Font Warning:|^LaTeX Warning: Reference.*undefined on input line|^Package hyperref Warning:/) {
+      $nc=" ";
+      $_=<>;
+      $_=<>;
+      while (<>) {
+	  if (! /^ *$/) {
+	      chomp;
+	      goto LINE;
+	  }
+      }
+  } else {
+      print "${nc}$_";
+      $nc="\n";
+  }
+}
+if ($nc ne "") {
+    print "\n";
+}
